tests/test_dse.py

```diff
@@ -2493,6 +2493,24 @@ def test_hoisting_pow_one(self):
 
         assert len([i for i in FindSymbols().visit(op) if i.is_Array]) == 0
 
+    @pytest.mark.parametrize('expr,expected', [
+        ('u.dy.dy', False),  # No STAR as hyperplane degenerates to 1D space
+        ('u.dx.dx + u.dy.dy', True),
+        ('u.dx.dx + u.dx.dy', True)
```

Comment:
Not sure Star is a very representative name, these are square stencil, `u.dx + u.dy` is what's called a star stencil

Reply:
nope, after CIRE they become stars -- that's the whole point of CIRE

Reply:
That's not possible. It either gets separated and becomes 1D lines or it's a square, there no possible way to compute `dx.dy` with a star stencil

Reply:
yes it is possible possible the u.dx is assigned to a temporary
so conceptually it's a square, but the implementation makes it a star

for me a square means that in a loop nest you have accesses sucha s `u[x+3][y+2]`. Not the case here

Reply:
Assigning it to a temporary means you are separating the dimensions. In which case you don't have a start you have a sequence of 1D line stencils

Reply:
yes, but if you take the two loop nests together, it's a star